Common/include/grid_movement_structure.hpp

```diff
@@ -1275,9 +1290,194 @@ class CVolumetricMovement : public CGridMovement {
 	 * \param[out] Number of iterations.
 	 */
 	unsigned long Get_nIterMesh(void);
+
+  /*!
+   * \brief Set the boundary dependencies in the mesh side of the problem
+   * \param[in] geometry - Geometrical definition of the problem.
+   * \param[in] config - Definition of the particular problem.
+   */
+  virtual void Boundary_Dependencies(CGeometry **geometry, CConfig *config);
 };
 
 /*! 
+ * \class CElasticityMovement
```

Comment:
As of now, they are just limited to FSI problems, basically because I didn't extend the boundary conditions to all of them. The objective is for it to be general - it resembles the FEA solver for linear problems, and therefore it is (in my opinion) more flexible than the original implementation. It needs a bit of extra work, that's why I haven't substitute the former implementation yet.